This script will look at your zipped fastq files (.fastq.gz) and make 1) a bam list and 2) a list of sample names that can be copied directly into the shell script. All fastq files must be in the same folder. Set the working directory to locate your fastq files. Change the quoted information in line 16 to the path where your bam files will be. Before beginning, make sure the fastq files are in the form: "genus-species-GNXXX_SX_L001_RX_001.fastq.gz". Otherwise, the code will not work correctly.


```{r}
setwd("F:/Mako") #set working directory for fastq

file.list<- list.files(pattern="R1_001\\.fastq\\.gz") #list file names

file.list<- sort(file.list) #sort file names by GN number
bam.names<- c()
sample.names<- c()
for (i in 1:length(file.list)){ #for each name
  file.trunc<- sub(".*?-.*?-(GN.*?_S.*?)_.*?_.*?_001.fastq.gz","\\1",file.list[i]) #extract info needed for bam list
  samp<- sprintf('"%s"', file.trunc) #add quotes
  sample.names<- append(sample.names,samp,after=length(sample.names)) #append sample names for shell script
  file.trunc2<- sprintf("/Volumes/ThunderBay/DD/melanopterus/analyses_exons+introns/local_realign/melanopterus/%s_R12_trimmed_q_15_sorted_rg_rmdup_localrealign.bam",file.trunc) #replace "" with path for bam files
bam.names<- append(bam.names, file.trunc2, after=(length(bam.names))) #append new name
}

fileConn<- file("bam.list", "wt")
for (j in 1:length(bam.names)){
  cat('', bam.names[j], sep="", file=fileConn)
  cat('\n', file=fileConn)
}
close(fileConn)


fileConn2<- file("sample.list", "wt")
cat('', sample.names, sep=" ", file=fileConn2)
close(fileConn2)


```

